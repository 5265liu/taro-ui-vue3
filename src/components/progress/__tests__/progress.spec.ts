import { mount } from '@vue/test-utils'
import AtProgress from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtProgress as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('Progress Snap', () => {
  it('should render initial Progress', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render Progress -- props percent', () => {
    const wrapper = factory({ percent: 25 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render Progress -- props isHidePercent', () => {
    const wrapper = factory({ isHidePercent: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render Progress -- props color', () => {
    const wrapper = factory({ color: '#FF4949' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render Progress -- props status(progress)', () => {
    const wrapper = factory({ status: 'progress' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render Progress -- props status(error)', () => {
    const wrapper = factory({ status: 'error' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render Progress -- props status(success)', () => {
    const wrapper = factory({ status: 'success' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
