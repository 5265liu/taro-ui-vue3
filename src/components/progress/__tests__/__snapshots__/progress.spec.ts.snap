// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Snap render Progress -- props color 1`] = `
<view
  class="at-progress"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 0%; height: 10px; background-color: rgb(255, 73, 73);"
      />
    </view>
  </view>
  <view
    class="at-progress__content"
  >
    0%
  </view>
</view>
`;

exports[`Progress Snap render Progress -- props isHidePercent 1`] = `
<view
  class="at-progress"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </view>
  </view>
  <!---->
</view>
`;

exports[`Progress Snap render Progress -- props percent 1`] = `
<view
  class="at-progress"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 25%; height: 10px;"
      />
    </view>
  </view>
  <view
    class="at-progress__content"
  >
    25%
  </view>
</view>
`;

exports[`Progress Snap render Progress -- props status(error) 1`] = `
<view
  class="at-progress at-progress--error"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </view>
  </view>
  <view
    class="at-progress__content"
  />
</view>
`;

exports[`Progress Snap render Progress -- props status(progress) 1`] = `
<view
  class="at-progress at-progress--progress"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </view>
  </view>
  <view
    class="at-progress__content"
  >
    0%
  </view>
</view>
`;

exports[`Progress Snap render Progress -- props status(success) 1`] = `
<view
  class="at-progress at-progress--success"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </view>
  </view>
  <view
    class="at-progress__content"
  />
</view>
`;

exports[`Progress Snap render initial Progress 1`] = `
<view
  class="at-progress"
>
  <view
    class="at-progress__outer"
  >
    <view
      class="at-progress__outer-inner"
    >
      <view
        class="at-progress__outer-inner-background"
        style="width: 0%; height: 10px;"
      />
    </view>
  </view>
  <view
    class="at-progress__content"
  >
    0%
  </view>
</view>
`;
